src/launchpad/parsers/android/dex/dex_mapping.py

```diff
@@ -0,0 +1,242 @@
+
+        return current_class, pending_file_name
+
+    def deobfuscate(self, obfuscated_class_name: str) -> str | None:
+        clazz = self._classes.get(obfuscated_class_name)
+        return clazz.deobfuscated_fqn if clazz else None
+
+    def deobfuscate_signature(self, obfuscated_signature: str) -> str | None:
+        clazz = self.lookup_obfuscated_signature(obfuscated_signature)
+        return clazz.deobfuscated_signature if clazz else None
+
+    def lookup_obfuscated_signature(self, obfuscated_signature: str) -> DexMappingClass | None:
+        obfuscated_fqn = AndroidCodeUtils.class_signature_to_fqn(obfuscated_signature)
+        return self.lookup_obfuscated_class(obfuscated_fqn)
+
+    def lookup_obfuscated_class(self, obfuscated_class_name: str) -> DexMappingClass | None:
+        return self._classes.get(obfuscated_class_name)
+
+    def lookup_deobfuscated_signature(self, deobfuscated_class_signature: str) -> DexMappingClass | None:
+        for clazz in self._classes.values():
+            if clazz.deobfuscated_signature == deobfuscated_class_signature:
+                return clazz
+        return None
+
+    def deobfuscate_method(self, class_name: str, obfuscated_method_name: str) -> str | None:
+        """Deobfuscate a method name for a given class.
+
+        Args:
+            class_name: The class name (can be obfuscated or deobfuscated)
+            obfuscated_method_name: The obfuscated method name
+
+        Returns:
+            The deobfuscated method name, or None if not found
+        """
+        # First try to find the class by obfuscated name
+        clazz = self.lookup_obfuscated_class(class_name)
+        if clazz is None:
+            # Try to find by deobfuscated signature
+            clazz = self.lookup_deobfuscated_signature(class_name)
+            if clazz is None:
+                # Try to match by deobfuscated FQN
+                for c in self._classes.values():
+                    if c.deobfuscated_fqn == class_name:
+                        clazz = c
+                        break
+                if clazz is None:
+                    return None
+
+        if clazz.methods is None:
+            return None
+
+        return clazz.methods.get(obfuscated_method_name)
+
+    def deobfuscate_field(self, class_name: str, obfuscated_field_name: str) -> str | None:
+        """Deobfuscate a field name for a given class.
+
+        Args:
+            class_name: The class name (can be obfuscated or deobfuscated)
+            obfuscated_field_name: The obfuscated field name
+
+        Returns:
+            The deobfuscated field name, or None if not found
+        """
+        # First try to find the class by obfuscated name
+        clazz = self.lookup_obfuscated_class(class_name)
+        if clazz is None:
+            # Try to find by deobfuscated signature
+            clazz = self.lookup_deobfuscated_signature(class_name)
+            if clazz is None:
```

Comment:
nit: could these branches be flattened? if i'm reading correctly they don't need to be nested.

Reply:
I don't see how it can, basically these are "fallbacks" if we can't find the deobfuscated class_name, each one is further trying to get us the proper deobfuscated class if the previous step failed